### PR TITLE
Add delly2

### PR DESCRIPTION
**Pull request by ihodes**
Waiting on it to run to make sure it's working

Fixes https://github.com/hammerlab/biokepi/issues/454

Fixes https://github.com/hammerlab/biokepi/issues/456

### CODE REVIEW

**Comment by ihodes** [comment]
@smondet This is verified to be working!

File: src/environment_setup/tool_providers.ml [hunk 1]
```diff
@@ -213,6 +213,18 @@ let freebayes =
         KEDSL.Program.(shf "export PATH=%s/bin/:$PATH" path))
     ~witness:(witness_list ["bin/freebayes"; "bin/bamleftalign"])
 
+let delly2 =
+  Installable_tool.make
+    Machine.Tool.Default.delly2
+    ~url:"https://github.com/dellytools/delly/releases/download/v0.7.7/delly_v0.7.7_parallel_linux_x86_64bit"
```
Comment (smondet) [inline src/environment_setup/tool_providers.ml]: Did you figure out what does the `parallel` mean here?

File: src/environment_setup/tool_providers.ml [hunk 2]
```diff
@@ -319,22 +331,31 @@ let kallisto =
...
   let url = "https://github.com/samtools/samtools/releases/download/1.3/\
              samtools-1.3.tar.bz2" in
-  let toplevel_tools = ["samtools"] in
   let htslib = ["bgzip"; "tabix" ] in
-  let tools = toplevel_tools @ htslib in
   let install_program ~path =
     let open KEDSL.Program in
     sh "make"
-    && shf "cp %s %s" (String.concat toplevel_tools ~sep:" ")  path
+    && shf "cp samtools %s"  path
```
Comment (smondet) [inline src/environment_setup/tool_providers.ml]: Why this change?

Comment (ihodes) [inline src/environment_setup/tool_providers.ml]: This I could definitely change back; I just didn't like the concat of 1 going on. Though now I repeat "samtools"…

File: src/environment_setup/tool_providers.ml (hunk 1 above)
Comment (ihodes) [inline src/environment_setup/tool_providers.ml]: "Delly2 supports parallel computing using the OpenMP API", so, since it works, might as well build this one depending on the backend one might use?

File: src/environment_setup/tool_providers.ml (hunk 2 above)
Comment (ihodes) [inline src/environment_setup/tool_providers.ml]: Fixed

**Comment by ihodes** [comment]
Back to you @smondet!
